File: programers_Lv2/이진 변환 반복하기.js
function solution(s) {
  const answer = [];

  let tempS = s;
  let count = 0;
  let zero = 0;

  while (tempS !== "1") {
    zero += [...tempS].filter((el) => el === "0").length;
    let rmZero = [...tempS].filter((el) => el === "1").join("");
    tempS = rmZero.length.toString(2);
    count++;
  }

  answer.push(count);
  answer.push(zero);

  return answer;
}

console.log(solution("110010101001"));
console.log(solution("1111111"));
console.log(solution("01110"));

File: programers_Lv2/이진 변환 반복하기.py
import re
def solution(s):
    answer = []

    temp_s = s
    count = 0
    zero = 0

    while temp_s != '1':
        zero += temp_s.count('0')
        rm_zero = re.sub('0', '', temp_s)
        temp_s = str(bin(len(rm_zero)))[2:]
        count += 1

    answer = [count, zero]

    return answer

print(solution("110010101001"))
print(solution("1111111"))
print(solution("01110"))